lib/auth/clt.go

```diff
@@ -2330,61 +2242,42 @@ type ProvisioningService interface {
 	RegisterNewAuthServer(token string) error
 }
 
+type Validation interface {
+	// ValidateTrustedCluster validates trusted cluster token with
+	// main cluster, in case if validation is successful, main cluster
+	// adds remote cluster
+	ValidateTrustedCluster(*ValidateTrustedClusterRequest) (*ValidateTrustedClusterResponse, error)
+}
+
 // ClientI is a client to Auth service
 type ClientI interface {
-	IdentityService
+	LocalClient
+	KeepAliver
+	WebService
+	Validation
+	WebAuth
+	LocalCluster
 	ProvisioningService
+	DomainNameGetter
+	CertAuthorityRotator
+	KeyGenerator
+
 	services.Trust
-	events.IAuditLog
-	events.Streamer
-	events.Emitter
 	services.Presence
 	services.Access
 	services.DynamicAccess
 	services.DynamicAccessOracle
-	WebService
-	session.Service
 	services.ClusterConfiguration
 	services.Events
 
-	types.WebSessionsGetter
-	types.WebTokensGetter
-
-	// NewKeepAliver returns a new instance of keep aliver
-	NewKeepAliver(ctx context.Context) (services.KeepAliver, error)
-
-	// RotateCertAuthority starts or restarts certificate authority rotation process.
-	RotateCertAuthority(req RotateRequest) error
-
-	// RotateExternalCertAuthority rotates external certificate authority,
-	// this method is used to update only public keys and certificates of the
-	// the certificate authorities of trusted clusters.
-	RotateExternalCertAuthority(ca services.CertAuthority) error
-
-	// ValidateTrustedCluster validates trusted cluster token with
-	// main cluster, in case if validation is successful, main cluster
-	// adds remote cluster
-	ValidateTrustedCluster(*ValidateTrustedClusterRequest) (*ValidateTrustedClusterResponse, error)
-
-	// GetDomainName returns auth server cluster name
-	GetDomainName() (string, error)
-
-	// GetClusterCACert returns the CAs for the local cluster without signing keys.
-	GetClusterCACert() (*LocalCAResponse, error)
+	events.IAuditLog
+	events.Streamer
+	events.Emitter
 
-	// GenerateServerKeys generates new host private keys and certificates (signed
-	// by the host certificate authority) for a node
-	GenerateServerKeys(GenerateServerKeysRequest) (*PackedKeys, error)
-	// AuthenticateWebUser authenticates web user, creates and  returns web session
-	// in case if authentication is successful
-	AuthenticateWebUser(req AuthenticateUserRequest) (services.WebSession, error)
-	// AuthenticateSSHUser authenticates SSH console user, creates and  returns a pair of signed TLS and SSH
-	// short lived certificates as a result
-	AuthenticateSSHUser(req AuthenticateSSHRequest) (*SSHLoginResponse, error)
+	session.Service
 
-	// ProcessKubeCSR processes CSR request against Kubernetes CA, returns
-	// signed certificate if successful.
-	ProcessKubeCSR(req KubeCSR) (*KubeCSRResponse, error)
+	types.WebSessionsGetter
+	types.WebTokensGetter
 
 	// Ping gets basic info about the auth server.
 	Ping(ctx context.Context) (proto.PingResponse, error)
```

Comment:
can you shuffle the below 4 methods (everything but `Ping`) into some of the embedded interfaces?